src/register.jl

```diff
@@ -1,12 +1,14 @@
 """
-    @register(expr, define_promotion, Ts = [Num, Symbolic, Real])
+    @register(expr, define_promotion = true, Ts = [Num, Symbolic, Real])
 
-Overload approperate methods such that ModelingToolkit can stop tracing into the
-registered function.
+Overload appropriate methods such that ModelingToolkit can stop tracing into the
+registered function. If `define_promotion` is true, then a promotion method is
```

Comment:
what the promotion method is doing needs to be defined a bit more. This promotion method is a promote_type of etc. etc.